libraries/base/Data/Monoid.hs

```diff
@@ -16,8 +16,42 @@
 -- Stability   :  experimental
 -- Portability :  portable
 --
--- A class for monoids (types with an associative binary operation that
--- has an identity) with various general-purpose instances.
+-- A type @a@ is a 'Monoid' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one, and a neutral
+-- element (`mempty`) such that 
+--
+-- > a <> mempty == mempty <> a == a
+--
+-- ==== __Examples__
+--
+-- The 'Sum' monoid is defined by the numerical addition operator and `0` as neutral element:
+--
+-- >>> mempty :: Sum Int
+-- Sum 0
+-- >>> Sum 1 <> Sum 2 <> Sum 3 <> Sum 4 :: Sum Int
+-- Sum {getSum = 10}
+--
+-- We can combine multiple values in a list into a single value using the `mconcat` function. 
+-- Note that we have to specify the type here since Int has more than one monoid instance:
+--
```

Comment:
Should be `'Int'`. I wouldn't say it *has" more than one "monoid instance". Perhaps "supports"?

Reply:
Better, perhaps: is a monoid under several different operations.

Reply:
That does indeed sound much better. Fixed!